### DISCUSSION TITLE
Lesson 4 Challenge - Stuck Contract Deployment transactions

### DISCUSSION THREAD

**Discussion by 0xTagger** [Cyfrin/foundry-full-course-cu#458] [Q&A]
Hi - The issue I am experiencing is deployed contracts being stuck pending or queued that I am unable to cancel. Below are the details.

I completed the Solve Challenge for Lesson 4 on Sepolia. Deploying my contract and using the getLatestData function went smooth with the Sepolia ETH/USD address from Chainlink Price Feed Addresses.
<img width="910" alt="Screenshot 2023-07-26 at 11 14 50 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/114359465/fb088015-de0b-49af-af26-ad3fa91d8928">

I then went to try to solve the challenge for Arbitrum by using the same contract in Remix only swapping out the Sepolia ETH/USD address with the ETH/USD mainnet address (0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419) in order to obtain the actual current ETH/USD price.

**ISSUE:** The contracts will not deploy and are stuck pending or queued. (I tried it twice) I am unable to manually cancel the transactions and am not sure what to do because there is no TxnHash for either one since they are a contract creation.  
<img width="346" alt="Screenshot 2023-07-26 at 11 25 31 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/114359465/bdcd9686-cf8c-4aa2-9a0a-cb93c09c147b">

<img width="1020" alt="Screenshot 2023-07-26 at 11 26 21 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/114359465/2abdab55-c962-4975-b997-d1753e80d2e6">

Any feedback would be much appreciated. Thanks.

EDIT: I went to advanced settings and cleared nonce data and my activity in Metamask. I am still unable to get a contract with the Ethereum mainnet ETH/USD price feed from Chainlink to deploy. Any feedback is welcomed.

**Comment by 0xwtfwtf** [comment]
I know you can cancel transactions on Metamask by sending a 0 ETH transaction wtih the same nonce and higher gas. You have to spend more gas sending this transaction than the transaction you're trying to cancel. [Here are the instructions - follow these not my description](https://support.ledger.com/hc/en-us/articles/9231137322525-Speed-up-or-cancel-pending-Ethereum-transaction-with-MetaMask-and-Ledger-device?docs=true)

Also, you could just start with a new address since it's a testnet.

Where are these challenges? I didn't know there were any.

**Reply by 0xTagger** [reply]
Yes that is true, but I would preferably like to keep using the same address to complete every challenge and also learn how to cancel them. Thank you for the response @0xwtfwtf  - I will try this. 

All challenges will show up as a QR code at the end of each lesson. They are all listed here as well --> https://github.com/Cyfrin/foundry-full-course-f23/discussions/135

**Reply by 0xwtfwtf** [reply]
Is there somewhere that explains what we do with these challenges? I tried to copy all the contracts into remix and compile and deploy them so I could do the twitter handle update but I ran into a problem when I copied the ABI into a library (it seems like the formatting is wrong) and imported it into each contract, and I don't know where to add the settings either. Can you explain how you did the first challenge?

**Reply by 0xTagger** [reply]
No, there is no where that explains what we do with the challenges. If you search Github for this course there are multiple posts with hints to help for each one, but no explanations or answers. You have to use the Write Contract option and Solve Challenge function with each challenge with the relative QR code or etherscan/arbiscan address.
